includes/admin/class-amp-editor-blocks.php

```diff
@@ -204,4 +205,131 @@ public function print_dirty_amp_scripts() {
 			wp_scripts()->do_items( $this->content_required_amp_scripts );
 		}
 	}
+
+	/**
+	 * Registers the dynamic block Latest Stories.
+	 * Much of this is taken from the Core block Latest Posts.
+	 *
+	 * @see register_block_core_latest_posts()
+	 */
+	public function register_block_latest_stories() {
+		register_block_type(
+			'amp/amp-latest-stories',
+			array(
+				'attributes'      => array(
+					'className'     => array(
+						'type' => 'string',
+					),
+					'storiesToShow' => array(
+						'type'    => 'number',
+						'default' => 5,
+					),
+					'order'         => array(
+						'type'    => 'string',
+						'default' => 'desc',
+					),
+					'orderBy'       => array(
+						'type'    => 'string',
+						'default' => 'date',
+					),
+				),
+				'render_callback' => array( $this, 'render_block_latest_stories' ),
+			)
+		);
+	}
+
+	/**
+	 * Renders the dynamic block Latest Stories.
+	 * Much of this is taken from the Core block Latest Posts.
+	 *
+	 * @see render_block_core_latest_posts()
+	 * @param array $attributes The block attributes.
+	 * @return string $markup The rendered block markup.
+	 */
+	public function render_block_latest_stories( $attributes ) {
+		$args        = array(
+			'post_type'        => AMP_Story_Post_Type::POST_TYPE_SLUG,
+			'posts_per_page'   => $attributes['storiesToShow'],
+			'post_status'      => 'publish',
+			'order'            => $attributes['order'],
+			'orderby'          => $attributes['orderBy'],
+			'suppress_filters' => false,
+			'meta_key'         => '_thumbnail_id',
+		);
+		$story_query = new WP_Query( $args );
+		$min_height  = $this->get_minimum_dimension( 'height', $story_query->posts );
+		$class       = 'amp-block-latest-stories';
+		if ( isset( $attributes['className'] ) ) {
+			$class .= ' ' . $attributes['className'];
+		}
+
+		ob_start();
+		?>
+		<div class="<?php echo esc_attr( $class ); ?>">
+			<ul class="latest-stories-carousel" style="height:<?php echo esc_attr( $min_height ); ?>px;">
```

Comment:
I realize that this contradicts what I said before about this being a `ul`, but isn't this actually supposed to be an `amp-carousel` outside of the editor?

Reply:
Sure, this could be an `<amp-carousel>` only on the front-end if you'd like. The idea of having the same `<ul>` carousel on the front-end and editor was to make this more DRY.

Still, there wouldn't be too much duplication in code by having the `<amp-carousel>` on the front-end. Just slight changes in markup, and some extra editor styles. Here's what it looks like when I just changed the `<ul>` to an `<amp-carousel>`:

<img width="730" alt="amp-carousel" src="https://user-images.githubusercontent.com/4063887/54064404-a85ebb00-41d9-11e9-80ad-fa4d16920294.png">

If you'd like to have an `<amp-carousel>` on the front-end, maybe this could pass an attribute to [ServerSideRender](https://wordpress.org/gutenberg/handbook/designers-developers/developers/components/server-side-render/#attributes) like `isEditor`, so that it's different in the editor and front-end.

Reply:
Yeah, that sounds good. As for DRY, we'll eventually only include `amp-carousel` so we can live with some WET for now.

Can the carousel be `fixed-height` so it can show multiple slides at the same time?

Reply:
Sure, https://github.com/ampproject/amp-wp/pull/1916/commits/1fb85db0899da34eaf265f8827c11f59c4cb8ef7 outputs a fixed-height `<amp-carousel>` on the front end. There are still styling fixes needed for the `<amp-carousel>`. If it's alright, I'll apply those tomorrow. Sorry I've been slow replying lately.

Reply:
**Some Styling Fixes Made**
*More To Come*

https://github.com/ampproject/amp-wp/pull/1916/commits/58bfd396f282ae7092d35604a0c8005907441ac2 makes some styling improvements, but some more are needed:
![current-carousel](https://user-images.githubusercontent.com/4063887/54076764-9bdb7080-4274-11e9-9dbc-7c7cb0fe3935.gif)